RS/YJ/Rule 6-13 (2022) (old: 6-12)

yunjoonjung-PNNL opened the pull request:
I haven't tested this rule yet.

jugonzal07 committed:
First cut at 6-12 rule test for ASHRAE 90.1-2022

yunjoonjung-PNNL committed:
Developed 6-12 rule

JacksonJ-KC commented:
FYI - This RDS was mistakenly labeled as a 2022 rule, but it should be in both 2019 and 2022.

jugonzal07 committed:
Added 6-12 to ASHRAE 2019 as well.

yunjoonjung-PNNL committed:
Merge branch 'RCT/YJ/rule_testing_2022' of https://github.com/pnnl/ruleset-checking-tool into RS/YJ/Rule_6-12

yunjoonjung-PNNL committed:
Fixed 6-12 rule

jugonzal07 committed:
Fixed issues in 6-12 a, b, and c. Some were related to test and standard descriptions. Test a missed are_scheduled_used_for_occupancy flags and c had it when it shouldn't have. This should be fixed now.

yunjoonjung-PNNL committed:
Merge pull request #1806 from pnnl/RT/JG/ruletest_6_12

Rule Test 6-12

yunjoonjung-PNNL committed:
Added 6-12 2019 rule

yunjoonjung-PNNL committed:
Merge branch 'feature/ashrae-9012022' of https://github.com/pnnl/ruleset-checking-tool into RS/YJ/Rule_6-12

JacksonJ-KC commented:
This is to become 6-13

yunjoonjung-PNNL committed:
Updated rule no. to 6-13

yunjoonjung-PNNL committed:
Fixed rule id

yunjoonjung-PNNL committed:
Merge branch 'feature/ashrae-9012022' of https://github.com/pnnl/ruleset-checking-tool into RS/YJ/Rule_6-12

weilixu commented on rct229/rulesets/ashrae9012019/__init__.py:
```diff
@@ -86,6 +86,7 @@
     "prm9012019rule66m62": "section6rule7",
     "prm9012019rule16x33": "section6rule8",
     "prm9012019rule22c86": "section6rule9",
+    "PRM9012019Rule86d29": "section6rule12",
```
Could we set this rule id to lower cases? Does not seem it follows the convention.

yunjoonjung-PNNL replied:
Good catch. Addressed!

yunjoonjung-PNNL committed:
Fixed rule id

yunjoonjung-PNNL committed:
Removed 6-12-2019 rules/testing JSON

yunjoonjung-PNNL committed:
Removed 6-12 from 2019 inti file

yunjoonjung-PNNL commented:
As discussed during the stand up, I've removed the 6-12-2019 rule.

JacksonJ-KC commented:
> As discussed during the stand up, I've removed the 6-12-2019 rule.

@weilixu @yunjoonjung-PNNL during the call earlier I did not visit the PR myself to confirm, sorry. Now I am seeing the rule language is in both 2019 and 2022 which explains my previous comment here. Do we want to renumber so that this rule has the same number in both rulesets?

yunjoonjung-PNNL committed:
Revert "Removed 6-12 from 2019 inti file"

This reverts commit fb0091c99ae81ae2b28cb44ec32c544c61623b00.

yunjoonjung-PNNL committed:
Revert "Removed 6-12-2019 rules/testing JSON"

This reverts commit 6601e9587305392c213df9f4b44b509c19746578.

yunjoonjung-PNNL committed:
Revert "Fixed rule id"

This reverts commit a7ff3975e2cc6acd5f495ee03e2cffba1d255ee1.

yunjoonjung-PNNL committed:
Added rule 6-13-2019

yunjoonjung-PNNL commented:
> > As discussed during the stand up, I've removed the 6-12-2019 rule.
> 
> @weilixu @yunjoonjung-PNNL during the call earlier I did not visit the PR myself to confirm, sorry. Now I am seeing the rule language is in both 2019 and 2022 which explains my previous comment here. Do we want to renumber so that this rule has the same number in both rulesets?

Thanks for double-checking. I've corrected the rule no. and can you check on this?

weilixu commented:
@yunjoonjung-PNNL I thought this rule is only in 2022?

yunjoonjung-PNNL committed:
Merge branch 'feature/ashrae-9012022' into RS/YJ/Rule_6-12

yunjoonjung-PNNL commented:
> FYI - This RDS was mistakenly labeled as a 2022 rule, but it should be in both 2019 and 2022.

@weilixu - Based on this comment, I added the same rule for 2019. Can @JacksonJ-KC confirm this?

JacksonJ-KC commented:
@weilixu Confirmed, this should be in both 2019 and 2022

weilixu commented:
@JacksonJ-KC I am confused - this rule number is 6-13, in 2019, we only have up to 6-9, I know there is a new 6-10 rule. If this rule should be in both 2019 and 2022, then should we renumber it to 6-11?

JacksonJ-KC commented:
I was thinking that too @weilixu which would trigger renumbers for a couple of other rules also under development for 2022. Maybe there is a simpler temporary solution to consider because I think other sections also need renumbers too (section 4, 10, 11)

JacksonJ-KC commented on rct229/rulesets/ashrae9012019/section6/section6rule13.py:
```diff
@@ -0,0 +1,120 @@
+                ),
+                each_rule=PRM9012019rule86d29.BuildingRule.SpaceRule(),
+                index_rmd=BASELINE_0,
+                list_path="$.building_segments[*].zones[*].spaces[*]",
```
add list filter to only include spaces that have interior lighting?

weilixu replied:
That is a good suggestion

yunjoonjung-PNNL replied:
Good point. Addressed.

JacksonJ-KC commented on rct229/rulesets/ashrae9012019/section6/section6rule13.py:
```diff
@@ -0,0 +1,120 @@
+            def get_calc_vals(self, context, data=None):
+                space_b = context.BASELINE_0
+
+                occupancy_sensor_controls_b = [
```
could use getattr_ here instead of manual_check_required

yunjoonjung-PNNL replied:
Addressed.

JacksonJ-KC commented on rct229/rulesets/ashrae9012019/section6/section6rule13.py:
```diff
@@ -0,0 +1,120 @@
+                    interior_lighting_b.get("occupancy_control_type")
+                    for interior_lighting_b in space_b["interior_lighting"]
+                ]
+                occupancy_sensor_schedules_b = [
```
could use getattr_ here instead of manual_check_required

yunjoonjung-PNNL replied:
Addressed.

JacksonJ-KC commented on rct229/rulesets/ashrae9012019/section6/section6rule13.py:
```diff
@@ -0,0 +1,120 @@
+            def rule_check(self, context, calc_vals=None, data=None):
+                occupancy_sensor_controls_b = calc_vals["occupancy_sensor_controls_b"]
+
+                return not any(
```
need to verify that all values in occupancy_sensor_schedules_b are True.

yunjoonjung-PNNL replied:
Addressed.

JacksonJ-KC replied:
Check this again. The `not any` was correct here before. There should be `and` to verify that all values in `occupancy_sensor_schedules_b` are True

yunjoonjung-PNNL replied:
Great - Corrected.

yunjoonjung-PNNL committed:
Addressed PR comments

yunjoonjung-PNNL commented:
@JacksonJ-KC - I've addressed your comments. Can you please check all were addressed correctly? The testing currently fails, but if the rule is correct, I'll reach out to @jugonzal07 for the testing file.

yunjoonjung-PNNL committed:
Corrected rule_check method

JacksonJ-KC reviewed (approved):
Approved!

weilixu commented:
Approved depending on the conflicts fix.

yunjoonjung-PNNL committed:
Merge branch 'feature/ashrae-9012022' of https://github.com/pnnl/ruleset-checking-tool into RS/YJ/Rule_6-12

weilixu committed:
Merge branch 'feature/ashrae-9012022' into RS/YJ/Rule_6-12
